apps/console/src/features/roles/components/edit-role/edit-role-permission.tsx

```diff
@@ -16,142 +16,429 @@
+     */
+    const searchAPIResources: DebouncedFunc<(query: string) => void> = 
+        useCallback(debounce((query: string) => {
+            setAPIResourceSearchQuery(`name co ${query}`);
+            mutateAPIResourcesListFetchRequest().finally(() => {
+                setAPIResourcesSearching(false);
+                setAPIResourceSearchQuery(undefined);
+            });
+        }, RoleConstants.DEBOUNCE_TIMEOUT), []);
+
+    /**
+     * Handles the selection of an API resource.
+     */
+    const onAPIResourceSelected = (event: SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
+        setSelectedAPIResourceId(data.value.toString());
+    };
+
+    /**
+     * Handles the change of the search query of application list.
+     */
+    const onSearchChangeAPIResources = (event: SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
+        setAPIResourcesSearching(true);
+        searchAPIResources(data?.searchQuery?.toString().trim());
+    };
+
+    const onChangeScopes = (apiResource: APIResourceInterface, scopes: ScopeInterface[]): void => {
+        const selectedScopes: SelectedPermissionsInterface[] = selectedPermissions.filter(
+            (selectedPermission: SelectedPermissionsInterface) => 
+                selectedPermission.apiResourceId !== apiResource.id
+        );
+        
+        selectedScopes.push(
+            {
+                apiResourceId: apiResource.id,
+                scopes: scopes
+            }
+        );
+
+        setSelectedPermissions(selectedScopes);
+    };
+
+    /**
+     * Handles the removal of an API resource.
+     */
+    const onRemoveAPIResource = (apiResourceId: string): void => {
+        // Remove the API resource from the selected API resources list.
+        setSelectedAPIResources(selectedAPIResources.filter((apiResource: APIResourceInterface) => {
+            return apiResource.id !== apiResourceId;
+        }));
+
+
+        // Removing the scopes(permissions) of the removed API resource from the selected permissions list.
+        setSelectedPermissions(selectedPermissions.filter((selectedPermission: SelectedPermissionsInterface) => {
+            return selectedPermission.apiResourceId !== apiResourceId;
+        }));
+    };
+
     return (
-        <div className="permissions-edit-container">
-            <PermissionList
-                isReadOnly={ isReadOnly }
-                isEdit={ !isReadOnly }
-                isRole
-                onSubmit={ onPermissionUpdate }
-                roleObject={ roleObject }
-                isSubmitting={ isSubmitting }
-            />
-        </div>
+        <EmphasizedSegment padded="very">
+            <Grid xs={ 8 }>
+                <Heading as="h4">
+                    Assigned Permissions
+                </Heading>
+                <Heading as="h6" color="grey" subHeading className="mb-5">
+                    Manage assigned permissions in the role.
+                    <DocumentationLink
```

Comment:
Do we need the learn more link for IS?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7